File: frontend/lecarden/src/app/shared/models/word.ts
import { Language } from './language';

export interface Word {
  id: number;
  name?: string;
  translation?: string;
  plural?: string;
  category?: string;
  imageUrl?: string;
  audioUrl?: string;
  builtIn?: boolean;
  userId?: number;
  languageId?: number;
  languageTO?: Language;
  example: string;
}

File: backend/filemanager/src/main/java/lecarden/filemanager/entity/Word.java
package lecarden.filemanager.entity;

import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Word {
    String name;
    String translation;
    String plural;
    String category;
    Long userId;
    Boolean builtIn;
    Long languageId;
}
